Code/C++/ModularArithmetic.cpp

```diff
@@ -0,0 +1,194 @@
+   Hence mod B with M-1 to reduce the number to a smaller one and then use power() method to compute (a^b)%m. 
+            
+   
+
+*/
+
+
+
+
+#include<bits/stdc++.h>
+using namespace std;
+
+typedef long long ll;
+typedef long double ld;
+typedef unsigned long long ull;
+#define MOD 1000000007
+#define ftio ios_base::sync_with_stdio(0); cin.tie(0); cout.tie(0); //This is just used for fast input output operations
+
+ll m,md;
+ll gcd(ll a,ll b,ll *x,ll *y); // Function declaration
+
+ll inverse(ll b,ll m){   // see point 1
+    ll x,y;
+    ll g=gcd(b,m,&x,&y);
+    if (g != 1) 
+        return -1;
+    
+    return (x%m + m) % m; 
+    
+}
+ll gcd(ll a, ll b, ll *x, ll *y) // Extended Euclidean algorithm to claculate GCD;
+{
+    if (a == 0) 
+    { 
+        *x = 0;
+        *y = 1; 
+        return b; 
+    } 
+  
+    ll x1, y1;  
+    ll g = gcd(b%a, a, &x1, &y1); 
+    *x = y1 - (b/a) * x1; 
+    *y = x1; 
+  
+    return g; 
+}
+ll gcd1(ll a, ll b) 
+{ 
+    if (b == 0) 
+        return a; 
+    return gcd1(b, a % b);  
+      
+} 
+
+ll modm(string s){ // see point 2
+     ll number = 0; 
+    for (ll i = 0; i < s.length(); i++) 
+    { 
+        number = (number*10 + (s[i] - '0')); 
+        number %= m; 
+    } 
+    return number; 
+}
+int main()
+{
+    ftio;
+    ll t;
+    cin>>t;
+    while(t--)
+    {
+        ll a,b;
+        string si,x,y;
+        cin>>si>>x>>y;
+        cin>>m;
+        
+        if(si=="+"){
+            a=modm(x);
+            b=modm(y);
+            cout<<((a+b) % m)<<"\n";
+        }
+        else if(si=="-"){
+           a=modm(x);
+            b=modm(y);
+            cout<<((a-b + m) % m)<<"\n"; // to handle the underflow, we have to add the mod value and then calculate the modulo operation
+        }
+        else if(si=="*"){
+           a=modm(x);
+            b=modm(y);
+            cout<<((a*b) % m)<<"\n";
+        }
+        else if(si=="^"){
+             ll res = 1;      
+             
+            a=modm(x);
+            md=m;
+            m-=1;   // see point 3
+            
+            b=modm(y);
+            m=md;
+            if (a == 0) cout<<0<<"\n"; 
+          else{
+            while (b > 0)  // calculating a^b after performing modular operation on them
+            {  
+                if (b & 1)  
+                    res = (res*a) % m;  
+                a = (a*a) % m;
+                b = b>>1; //Similar to  b=b/2
+                  
+            }
+            cout<<(res%m)<<"\n";
+          }    
+        }
+        else{
+            a=modm(x);
+            b=modm(y);
+            ll g = gcd1(a,b);
+            if(g!=1){
+                a=a/g;
+                b=b/g;
+            }
+            ll inv = inverse(b, m); // see the explaination above 
+            if (inv == -1) 
+               cout <<-1<<"\n"; 
+            else
+               cout<< (a * inv) % m <<"\n";
+        }
+        
+        
+
+        
```

Comment:
Remove extra white space.

Reply:
@yasharth291 I am trying, due to exams I was inactive, apologizes for this. Please add this problem. I will try to update that.Please.
@OjusWiZard Please.